src/env.ts

```diff
@@ -0,0 +1,135 @@
+import envSchema from "env-schema";
+
+interface Env {
+  API_HOST: string;
+  API_PORT: number;
+
+  PGHOST: string;
+  PGPORT: number;
+  PGUSER: string;
+  PGPASSWORD: string;
+  PGDATABASE: string;
+  BLOCKCHAIN_API_PGHOST: string;
+  BLOCKCHAIN_API_PGPORT: number;
+  BLOCKCHAIN_API_PGUSER: string;
+  BLOCKCHAIN_API_PGPASSWORD: string;
+  BLOCKCHAIN_API_PGDATABASE: string;
+
+  STACKS_NODE_RPC_HOST: string;
+  STACKS_NODE_RPC_PORT: number;
+
+  METADATA_STRICT_MODE: boolean;
+  METADATA_MAX_RETRIES: number;
+  /**
+   * The max number of immediate attempts that will be made to retrieve metadata from external URIs
+   * before declaring the failure as a non-retryable error.
+   */
+  METADATA_MAX_IMMEDIATE_URI_RETRIES: number;
+  METADATA_FETCH_TIMEOUT_MS: number;
+  /**
+   * The maximum number of bytes of metadata to fetch. If the fetch encounters more bytes than this
+   * limit it throws and the metadata is not processed.
+   */
+  METADATA_MAX_PAYLOAD_BYTE_SIZE: number;
+
+  JOB_QUEUE_CONCURRENCY_LIMIT: number;
+  JOB_QUEUE_SIZE_LIMIT: number;
+}
+
+export function getEnvVars(): Env {
+  const schema = {
```

Comment:
This current schema definition is nice and minimalistic and we can keep it that way, but typebox could be used to define this schema in a type-safe way if we want.